VMD PCH rootbus

sdurawa opened the pull request:
Add PCH rootbus support

mtkaczyk commented on drivers/pci/controller/vmd.c:
```diff
@@ -34,6 +34,18 @@
+enum vmd_resource {
+	VMD_RES_CFGBAR = 0,
+	VMD_RES_MBAR_1,
+	VMD_RES_MBAR_2,
```
```suggestion
/*VMD Resource MemoryBar 1 */
VMD_RES_MBAR_1,
```

mtkaczyk replied:
And same for `MBAR_2` - just to explain full meaning.

mtkaczyk reviewed (approved):
LGTM, one small suggestion

mtkaczyk commented on drivers/pci/controller/vmd.c:
```diff
@@ -790,6 +834,27 @@ static void vmd_configure_cfgbar(struct vmd_dev *vmd)
 		.flags = IORESOURCE_BUS | IORESOURCE_PCI_FIXED,
 	};
+
+	if (vmd_has_pch_rootbus(vmd)) {
```
please put declarations inside if:
```suggestion
		u16 ioc_range = 0;
		u16 pch_range = 0;

		pci_read_config_word(vmd->dev, VMD_BUSRANGE0, &ioc_range);
		pci_read_config_word(vmd->dev, VMD_BUSRANGE1, &pch_range);
```

mtkaczyk reviewed (approved):
PATCH v2 LGTM, one small suggestion

mtkaczyk commented on drivers/pci/controller/vmd.c:
```diff
@@ -404,8 +404,21 @@ static inline u8 vmd_has_pch_rootbus(struct vmd_dev *vmd)
+	 */
+	bus_number = (vmd_has_pch_rootbus(vmd) &&
+		      bus->number == VMD_PRIMARY_PCH_BUS) ?
+			     vmd->busn_start[VMD_BUS_1] : bus->number;
```
formatting here
```suggestion
	      vmd->busn_start[VMD_BUS_1] : bus->number;
```

mtkaczyk reviewed (commented):
one formatting issue

Commit:
PCI: vmd: Clean up vmd_enable_domain function

This function is too long and needs to be shortened to make it more readable.
This clean up is a prework for enablement additional VMD bus range.
It doesn't change functional behavior of vmd_enable_domain().

Suggested-by: Nirmal Patel <nirmal.patel@linux.intel.com>
Reviewed-by: Mariusz Tkaczyk <mariusz.tkaczyk@linux.intel.com>
Signed-off-by: Szymon Durawa <szymon.durawa@linux.intel.com>

Commit:
PCI: vmd: Add WA for VMD PCH rootbus support

VMD PCH rootbus primary number is 0x80 and pci_scan_bridge_extend()
cannot assign it as "hard-wired to 0" and marks setup as broken. To
avoid this, PCH bus number has to be the same as PCH primary number.

Suggested-by: Nirmal Patel <nirmal.patel@linux.intel.com>
Reviewed-by: Mariusz Tkaczyk <mariusz.tkaczyk@linux.intel.com>
Signed-off-by: Szymon Durawa <szymon.durawa@linux.intel.com>

Commit:
PCI: vmd: Add VMD PCH rootbus support

Starting from Intel Arrow Lake VMD enhacement introduces separate
rotbus for PCH. It means that all 3 MMIO BARs exposed by VMD are
shared now between CPU IOC and PCH. This patch adds PCH bus
enumeration and MMIO management for devices with VMD enhancement
support.

Suggested-by: Nirmal Patel <nirmal.patel@linux.intel.com>
Reviewed-by: Mariusz Tkaczyk <mariusz.tkaczyk@linux.intel.com>
Signed-off-by: Szymon Durawa <szymon.durawa@linux.intel.com>
